wasm_libs/build_gsl.sh

```diff
@@ -3,7 +3,18 @@ command -v emcc >/dev/null 2>&1 || { echo "Script requires emcc but it's not ins
 cd "${0%/*}"
 if ! [[ $(find gsl-2.6.tar.gz -type f 2>/dev/null && md5sum -c gsl.md5 &>/dev/null) ]]; then
     echo "Fetching GSL 2.6"
-    wget http://ftpmirror.gnu.org/gsl/gsl-2.6.tar.gz
+    retry_count=0
+    max_retries=2
+    while (( retry_count < max_retries )); do
+        wget http://ftpmirror.gnu.org/gsl/gsl-2.6.tar.gz && break
```

Comment:
if I visit gnu.org and let it redirect me to the nearest mirror to get a copy of gsl, it is https://mirror.ossplanet.net/gnu/gsl/. So maybe if http://ftpmirror.gnu.org/gsl/gsl-2.6.tar.gz is not accessible, we can try this mirror?

Reply:
or this is not just a wget from Taiwan?

Reply:
@kswang1029 Trying different mirrors is an excellent idea! I know it says [here](https://www.gnu.org/prep/ftp.html) that by going to http://ftpmirror.gnu.org, it automatically chooses a nearby up-to-date mirror. I didn't know that before. I have just checked with `wget http://ftpmirror.gnu.org/gsl/gsl-2.6.tar.gz` and it used `mirror.ossplanet.net`.
So maybe to be even safer, we should explicitly try a completely different mirror too.

How about the following change: I have kept the retry part in all four of the download scripts. However, to the GSL script, I have added the fallback to try to download from ossplanet once, plus an extra fallback to download from a mirror in Berkeley.

I have double checked the md5sum of the gsl-2.6.tar.gz from ossplanet and Berkeley and they agree with the md5sum we have on file.

Reply:
yes, having a few explicit mirror sites to try is a good idea. We may consider mirrors in Japan and US perhaps?

Reply:
OK. I have added two more explicit mirror sites. Now it will try Taiwan > Japan > US (Berkeley) > Europe (Germany). It should definitely be able to succeed every time now.